xsd/common.xsd

```diff
@@ -224,44 +227,564 @@
 
 	<xs:element name="modifier">
 		<xs:complexType>
-			<xs:choice minOccurs="0" maxOccurs="unbounded">
-				<xs:choice maxOccurs="unbounded">
-					<xs:element ref="cost" />
-					<xs:element ref="skill_bonus" />
-					<xs:element ref="attribute_bonus" />
-					<xs:element name="name" type="StringWithCompareAttribute" />
-					<xs:element name="notes" type="StringWithCompareAttribute" />
-					<xs:element name="levels" type="xs:unsignedInt" />
-					<xs:element name="reference" type="xs:string" />
-					<xs:element name="affects" type="xs:string" />
-				</xs:choice>
+			<xs:choice maxOccurs="unbounded">
+				<xs:element ref="cost" />
+				<xs:element ref="skill_bonus" />
+				<xs:element ref="attribute_bonus" />
+				<xs:element name="name" type="StringWithCompareAttribute" />
+				<xs:element name="notes" type="StringWithCompareAttribute" />
+				<xs:element name="levels" type="xs:unsignedInt" />
+				<xs:element name="reference" type="xs:string" />
+				<xs:element name="affects" type="xs:string" />
 			</xs:choice>
-			<xs:attribute name="version" type="xs:unsignedShort" use="optional" />
-			<xs:attribute name="enabled" type="xs:string" use="optional" />
+			<xs:attribute name="version" type="xs:unsignedShort" />
+			<xs:attribute name="enabled">
+				<xs:simpleType>
+					<xs:restriction base="xs:string">
+						<xs:enumeration value="yes"/>
+						<xs:enumeration value="no"/>
+					</xs:restriction>
+				</xs:simpleType>
+			</xs:attribute>
 		</xs:complexType>
 	</xs:element>
 
+	<xs:simpleType name="YesNo">
+		<xs:restriction base="xs:string">
+			<xs:enumeration value="yes"/>
+			<xs:enumeration value="no"/>
+		</xs:restriction>
+	</xs:simpleType>
+
+	<xs:complexType name="Reference">
+		<xs:annotation>
+			<xs:documentation>
+				This isn't used anywhere...it's offered as a suggestion for how references could be presented.
+				Zero, one, or multiple occurences allowed.
+			</xs:documentation>
+		</xs:annotation>
+		<xs:attribute name="Book" type="xs:string" use="required" />
+		<xs:attribute name="Page" type="xs:unsignedShort" />
+	</xs:complexType>
+
+	<xs:element name="categories">
+		<xs:complexType>
+			<xs:choice maxOccurs="unbounded">
+				<xs:element name="category" type="xs:string" />
+			</xs:choice>
+		</xs:complexType>
+	</xs:element>
+
+	<xs:simpleType name="CategoryName">
+		<xs:restriction base="xs:string">
```

Comment:
This restriction is not practical. Categories are entirely free-form and subject to the input that the user does. This might reflect the data at this moment in time, but will become incorrect very rapidly.